libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -0,0 +1,623 @@
+from __future__ import annotations
+
+import enum
+import logging
+import uuid
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+)
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.vectorstores.base import VectorStore
+
+
+class DistanceStrategy(str, enum.Enum):
+    """Enumerator of the Distance strategies."""
+
+    EUCLIDEAN = "euclidean"
+    COSINE = "cosine"
+
+
+DEFAULT_DISTANCE_STRATEGY = DistanceStrategy.COSINE
+
+
+class Neo4jVector(VectorStore):
+    """`Neo4j` vector index.
+
+    To use, you should have the ``neo4j`` python package installed.
+
+    Args:
+        url: Neo4j connection url
+        username: Neo4j username.
+        password: Neo4j password
```

Comment:
can you add the neo4j database as optional arg, defaulting to "neo4j" ?

Reply:
that's already there, there are a bunch of optional parameters I left from the docstring